tools/wake-format/emitter.cpp

```diff
@@ -524,19 +526,335 @@ wcl::doc Emitter::walk_ascribe(ctx_t ctx, CSTElement node) {
   MEMO_RET(walk_placeholder(ctx, node));
 }
 
+std::vector<CSTElement> collect_binary_parts(CSTElement collect_over, CSTElement node) {
+  if (node.id() != CST_BINARY) {
+    return {node};
+  }
+
+  // WARNING: I'm using Node() here which is differnt than everywhere else
+  // This is probably not right since I'm loosing comments
+  CSTElement left = node.firstChildNode();
+  CSTElement op = left;
+  op.nextSiblingNode();
+  CSTElement right = op;
+  right.nextSiblingNode();
+
+  if (!(op.id() == CST_OP && op.firstChildElement().id() == collect_over.id() &&
+        op.firstChildElement().fragment().segment().str() ==
+            collect_over.fragment().segment().str())) {
+    return {node};
+  }
+
+  auto left_collect = collect_binary_parts(collect_over, left);
+  auto right_collect = collect_binary_parts(collect_over, right);
```

Comment:
This is correct for a kind of strange reason IMO. This algorithm wouldn't work with an AST for instance. It only works becasue a PAREN will get in the way and the parser will save you from having to deal with the case where both the left and the right side are the collect_over operator. I would instead suggest separate left collect and right collect implementations personally but I'm willing to be convinced otherwise.

Reply:
I'm down to fix this, but can I delay until after the next PR to minimize merge conflicts / simplify the impl?

Reply:
nvm, I managed to pull it in